File: obj-c/Point.m
/* Implementation.  Yay!
 *
 * Simon Heath
 * 14/6/2003
 */

#import "Point.h"

@implementation Point

+ new
{
   return [super new];
}

- init
{
   x = 0;
   y = 0;
   return self;
}

- (int) getX
{
   return x;
}

- (int) getY
{
   return y;
}

- (void) moveTo: (int) newx: (int) newy
{
   x = newx;
   y = newy;
}

- (void) print
{
   printf( "(%d, %d)\n", x, y );
}

@end

File: rust/point.rs
struct Point {
  x: int,
  y: int,
}

fn main() {
  let origin = Point{x: 0i, y: 0i};
  println!("The origin is at ({}, {})", origin.x, origin.y)

  let mut point = Point{x: 0i, y: 0i};
  point.x = 6;
  println!("The point is at ({}, {})", point.x, point.y);
}
